tests/all-tests.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+var fluid = fluid || require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.test.mail.smtp.testSuite");
+
+require("../src/js/index.js");
+
+var jqUnit = fluid.require("jqUnit");
+var fs     = require("fs");
+
+gpii.test.mail.smtp.testSuite.mailOptions = {
+    from:    "sender@localhost",
+    to:      "recipient@localhost",
+    subject: "Test Subject",
+    text:    "Test Body"
+};
+
+gpii.test.mail.smtp.testSuite.isSaneResponse = function (error, info) {
+    jqUnit.assertNull("There should be no mail errors", error);
+
+    jqUnit.assertNotNull("There should message info returned...", info);
+    if (info) {
+        jqUnit.assertNotNull("There should be a message ID", info.messageId);
+        jqUnit.assertEquals("There should be an accepted message...", 1, info.accepted.length);
+        jqUnit.assertEquals("There should be no rejected messages...", 0, info.rejected.length);
+        jqUnit.assertEquals("The sender should be correct", gpii.test.mail.smtp.testSuite.mailOptions.from, info.envelope.from);
+        jqUnit.assertEquals("The recipient should be correct", gpii.test.mail.smtp.testSuite.mailOptions.to, info.envelope.to[0]);
+    }
+};
+
+gpii.test.mail.smtp.testSuite.mailTest = function (callback) {
+    fluid.registerNamespace("gpii.test.mail.smtp.testInstance");
```

Comment:
Should not register namespaces from within code. The callback should not be configured this way but instead as an event listener with a global name - the function can then be defined as a standard globally namespaced function. The configuration given to the constructor can then be written as a standard grade.